proposal to use the term "format" (#83)
mjoras/SCONE-PROTOCL: documents/charter.md

The proposed deliverables for SCONEPRO are as follows:

> * Develop a Standards Track specification describing a SCONEPRO protocol to communicate network properties applicable to streaming video from the SCONEPRO monitor to the client(s), and specifying a format for those applicable network properties.

Reviewer: I'm not sure that this needs to separate protocol and format.  A format is a protocol (though it is often not a complete protocol), so the extra words could go.

```suggestion
* Develop a Standards Track specification describing a SCONEPRO protocol to communicate network properties applicable to streaming video from the SCONEPRO monitor to the client(s).
```

Reviewer 2: @martinthomson - you may very well be right, but let me try to explain what I'm trying to make clear, so we'll know whether I'm being unclear, or just wrong. 

Tl;dr

Is this better than the current text in the PR?

> * Develop a Standards Track specification describing how a set of network properties applicable to streaming video will be transferred from a SCONEPRO monitor to one or more clients, what network properties will be included in this set, and how each of the network properties in the set will be encoded and decoded

If not, please keep reading.

- We need a description of a protocol that transfers a blob of applicable network properties that are relevant to streaming video from the SCONEPRO monitor to one or more clients, without reference to what those network properties are, or how they are encoded.
- We need a description of the applicable network properties that will appear inside the blob, and how they will be encoded by the SCONEPRO monitor and decoded by the client(s). 

_(My goal is that if we identify additional applicable network properties to include in the blob, the description of the set of applicable network properties in the blob would change, but the description of how the set of applicable network properties are placed in the blob and transferred would **not** change. Does that make sense?)_

A previous version of the proposed charter placed these two things into two deliverables (so they weren't muddled). We decided to combine these two things into one deliverable, and @mirjak pointed out that my first attempt at the description of that deliverable muddled these two things. I'm trying to un-muddle these two things, while still describing a single, combined deliverable.

Reviewer 2: @martinthomson - could you let us know if [my comment](https://github.com/mjoras/SCONE-PROTOCL/pull/83#discussion_r1666964258) makes sense?

Reviewer: So these are all too many words.  A protocol is what we're describing, so using that word is fine, but the idea that there is a distinction between "protocol" and "format" makes no sense to me.  You would only define a discrete format if you had some intention to use that outside of the protocol.

Yes, we need to know what the signal looks like (which is still an unsolved question).  But it is enough to say that the group will define a standards track specification describing how a network element might communicate about constraints on streaming video throughput to clients.

    * connection establishment, communication signaling and data format (container to share network information)
    * specify and justify privacy and security requirements

> * Develop a standard track specification based on SCONEPRO protocol to specify a format for network properties applicable to streaming video to be communicated from the SCONEPRO monitor to the client(s). Its contents might include

Reviewer 2: @mirjak, thanks for submitting this. IMO, the problem is that we merged two deliverables (one for a protocol, the other for the format), and I was paying attention to the sub-bullet level of detail, but didn't notice that the bullet-level of detail was incomplete. 

I think the correct change would be this.

```suggestion
* Develop a standard track specification describing a SCONEPRO protocol to communicate network properties applicable to streaming video from the SCONEPRO monitor to the client(s), and specifying a format for those applicable network properties.  Its contents might include
```
